xbmc/guilib/GUIFont.cpp

```diff
@@ -98,24 +102,26 @@ void CGUIFont::DrawText(float x,
   maxPixelWidth = ROUND(static_cast<double>(
       maxPixelWidth / CServiceBroker::GetWinSystem()->GetGfxContext().GetGUIScaleX()));
   std::vector<UTILS::COLOR::Color> renderColors;
-  for (unsigned int i = 0; i < colors.size(); i++)
-    renderColors.push_back(CServiceBroker::GetWinSystem()->GetGfxContext().MergeAlpha(colors[i] ? colors[i] : m_textColor));
-  if (!shadowColor) shadowColor = m_shadowColor;
+  for (const auto& color : colors)
+    renderColors.emplace_back(
+        CServiceBroker::GetWinSystem()->GetGfxContext().MergeAlpha(color ? color : m_textColor));
```

Comment:
GetWinSystem is a pointer, therefore has a possibility to be null. Really should check for null imo

Reply:
To make for here it becomes to heavy, by the really big amount of `CServiceBroker::GetWinSystem()` calls must be there a another way, instead of a `if (null)`.

Reply:
essentially you cant chain the ->function into the getwinsystem, as it'll just crash if a null ptr is returned by the getwinsystem (ie the call is then nullptr->GetGfcContext() = crash).

Also keep in mind, this was all brought about due to a headless future possibly. if we have no win system, why would we want to blindly run gui related stuff, therefore if getwinsystem == nullptr, we move on without processing, hence the checks

Reply:
@fuzzard have added on last commit a pointer check about, can you look over it?